File: algorithm/力扣热题100/11_盛最多水的容器.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    let i = 0, j = height.length - 1;
    let res = 0;
    while (i < j) {
        res = Math.max(res, (j - i) * Math.min(height[i], height[j]))
        if (height[i] < height[j]) {
            i++
        } else {
            j--
        }
    }
    return res
};

File: algorithm/力扣热题100/11_盛最多水的容器.go
package main

func maxArea(height []int) int {
	i := 0
	j := len(height) - 1
	res := 0
	for i < j {
		res = max(res, (j-i)*min(height[i], height[j]))
		if height[i] < height[j] {
			i++
		} else {
			j--
		}
	}
	return res
}
